### DISCUSSION TITLE
Site deployment

### DISCUSSION THREAD

**Discussion by patrykpalej** [cotes2020/jekyll-theme-chirpy#1466] [Q&A]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### What is the topic?

Sharing tips and tricks

### Description

Hello. I just set up my repo with customized version of chirpy theme which is available here:

https://github.com/patrykpalej/patrykpalej.github.io

The website domain is:
https://patrykpalej.dev/

The problem is that after entering the website, I've got 404 from github pages. It can correctly redirect from .github.io to my domain, but my website doesn't show.

I have two branches - master and gh-pages. I didn't modify the default github action script, should I? 2 out of 4 CI/CD jobs failed because of:

```
In Gemfile:
  jekyll-theme-chirpy was resolved to 6.3.1, which depends on
    jekyll-archives was resolved to 2.2.1, which depends on
      jekyll was resolved to 4.3.3, which depends on
        jekyll-sass-converter was resolved to 3.0.0, which depends on
          sass-embedded was resolved to 1.69.7, which depends on
            google-protobuf
Error: The process '/opt/hostedtoolcache/Ruby/3.3.0/x64/bin/bundle' failed with exit code 5
```

**Comment by joshjohanning** [comment]
@patrykpalej change [this line](https://github.com/patrykpalej/patrykpalej.github.io/blob/master/.github/workflows/pages-deploy.yml#L45) from `3` to `3.2`

Reference:

https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/.github/workflows/pages-deploy.yml.hook#L45

It looks like using `3` now uses Ruby `3.3.0`, which appears to be too new for what Chirpy is expecting. 

see also:
- https://github.com/cotes2020/jekyll-theme-chirpy/issues/1429

**Reply by patrykpalej** [reply]
Thanks, I'll try that. And do you know whether I should put anything on `gh-pages` branch or it should be automated? I saw on other repos that there is content of _site folder but I don't know if I should put it manually.

**Reply by joshjohanning** [reply]
@patrykpalej I would recommend setting your GitHub Pages to use the "GitHub Actions" as the source instead of "from branch"

![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/19912012/0b1cd8bf-5d86-4132-a8ef-f05e733041c2)

Then you can use the workflow here (without the `.hook` at the end of the file extension)

https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/.github/workflows/pages-deploy.yml.hook

Example of mine: https://github.com/joshjohanning/joshjohanning.github.io/blob/main/.github/workflows/pages-deploy.yml

**Reply by patrykpalej** [reply]
Actually I set up the build from Github Actions and used the same script as yours

https://github.com/patrykpalej/patrykpalej.github.io/blob/master/.github/workflows/pages-deploy.yml

But how does it know to create gh-pages branch? It's not mentioned there.

After all, my pipelines fail unfortunately :/
![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/34798604/8c3f689c-a52c-4bd0-8218-26bc83f738b6)

**Reply by cotes2020** [reply]
Hi @patrykpalej, Chirpy `6.4.0` has a bug in generating image URLs, please use the latest version [`6.4.1`](https://github.com/cotes2020/jekyll-theme-chirpy/releases/tag/v6.4.1).

**Reply by patrykpalej** [reply]
I cloned the starter some time ago and I have 6.3.1 version. Is it good enough?

`gem "jekyll-theme-chirpy", "~> 6.3", ">= 6.3.1"`  (Gemfile)

**Reply by joshjohanning** [reply]
Looks like your most recent build was able to deploy the site 🎉   Looks like you used the branch method [here](https://github.com/patrykpalej/patrykpalej.github.io/actions/runs/7480668374) though.

It looks like the Actions method [failed the first time ](https://github.com/patrykpalej/patrykpalej.github.io/actions/runs/7479903667/job/20358108279)with the test step because it couldn't validate the assets on the domain since they didn't exist yet (no valid deployment). 

Not sure if this is a timing issue that occurs on first deployment or not.  You could commented out the `test site` step to bypass

```yml
      # - name: Test site
      #   run: |
      #     bundle exec htmlproofer _site \
      #       \-\-disable-external=true \
      #       \-\-ignore-urls "/^http:\/\/127.0.0.1/,/^http:\/\/0.0.0.0/,/^http:\/\/localhost/"
```

Now that you have your site, I'm guessing you could re-enable GitHub Actions deployment and it would work now since the failing assets would now live at your domain.

**Reply by patrykpalej** [reply]
> Looks like your most recent build was able to deploy the site 🎉 Looks like you used the branch method [here](https://github.com/patrykpalej/patrykpalej.github.io/actions/runs/7480668374) though.

Yeah, I tried many different approaches and finally switched to deployment from branch. Unfortunately it didn't update the `gh-pages` branch after pushing changes so I kept playing around and it's messed up again. 

Actually, for now the website is available, but it doesn't have the correct styles and links are invalid. Instead of adding relative path to the current, they add the whole absolute path to the current path. Also, the page is not available under `.github.io` domain, which should redirect to the main domain.

Maybe that's because I build the website locally and then push it to the Github.

**Reply by patrykpalej** [reply]
I'm not quite sure what happened but after commenting `url` and `baseurl` in _config.yml everything works fine (I hope so). Does it make sense?

**Reply by cotes2020** [reply]
Please take a few minutes to check out the [deployment tutorial](https://chirpy.cotes.page/posts/getting-started/#deployment) that will answer all your questions.
